docs/proposals/04-new-core-gardener-cloud-apis.md

```diff
@@ -0,0 +1,698 @@
+    #   oidcConfig:
+    #     caBundle: |
+    #       -----BEGIN CERTIFICATE-----
+    #       Li4u
+    #       -----END CERTIFICATE-----
+    #     clientID: client-id
+    #     groupsClaim: groups-claim
+    #     groupsPrefix: groups-prefix
+    #     issuerURL: https://identity.example.com
+    #     usernameClaim: username-claim
+    #     usernamePrefix: username-prefix
+    #     signingAlgs: RS256,some-other-algorithm
+    #-#-# only usable with Kubernetes >= 1.11
+    #     requiredClaims:
+    #       key: value
+    #   admissionPlugins:
+    #   - name: PodNodeSelector
+    #     config: |
+    #       podNodeSelectorPluginConfig:
+    #         clusterDefaultNodeSelector: <node-selectors-labels>
+    #         namespace1: <node-selectors-labels>
+    #         namespace2: <node-selectors-labels>
+    #   auditConfig:
+    #     auditPolicy:
+    #       configMapRef:
+    #         name: auditpolicy
+    # kubeControllerManager:
+    #   featureGates:
+    #     SomeKubernetesFeature: true
+    #   horizontalPodAutoscaler:
+    #     syncPeriod: 30s
+    #     tolerance: 0.1
+    #-#-# only usable with Kubernetes < 1.12
+    #     downscaleDelay: 15m0s
+    #     upscaleDelay: 1m0s
+    #-#-# only usable with Kubernetes >= 1.12
+    #     downscaleStabilization: 5m0s
+    #     initialReadinessDelay: 30s
+    #     cpuInitializationPeriod: 5m0s
+    # kubeScheduler:
+    #   featureGates:
+    #     SomeKubernetesFeature: true
+    # clusterAutoscaler:
+    #   scaleDownUtilizationThreshold: 0.5
+    #   scaleDownUnneededTime: 30m
+    #   scaleDownDelayAfterAdd: 60m
+    #   scaleDownDelayAfterFailure: 10m
+    #   scaleDownDelayAfterDelete: 10s
+    #   scanInterval: 10s
+    worker:
+    - name: cpu-worker
+      minimum: 3
+      maximum: 5
+    # maxSurge: 1
+    # maxUnavailable: 0
+      machine:
+        type: m5.large
+        image:
+          name: <some-os-name>
+          version: <some-os-version>
+        # providerConfig:
+        #   <some-os-specific-configuration>
+      volume:
+        type: gp2
+        size: 20Gi
+    # providerConfig:
+    #   <some-provider-specific-worker-config>
+    # labels:
+    #   key: value
+    # annotations:
+    #   key: value
+    # taints: # See also https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+    # - key: foo
+    #   value: bar
+    #   effect: NoSchedule
+    # kubernetes:
+    #   kubeProxy:
+    #     featureGates:
+    #       SomeKubernetesFeature: true
+    #     mode: IPVS
+    #   kubelet:
+    #     cpuCFSQuota: true
+    #     cpuManagerPolicy: none
+    #     podPidsLimit: 10
+    #     featureGates:
+    #       SomeKubernetesFeature: true
+      zones:
+      - europe-central-1a
+      - europe-central-1b
+  kubernetes:
+    version: 1.15.1
+  # allowPrivilegedContainers: true # 'true' means that all authenticated users can use the "gardener.privileged" PodSecurityPolicy, allowing full unrestricted access to Pod features.
+  dns:
```

Comment:
Concerning my previous comment on `seed.dns`:

We can take the following structure:

```
# optional - assigns default domain with default provider if not specified.
dns:
  domain: seed1.dev.example.com
  provider: <provider name> # optional - uses default provider if non is given.
```

And let the user create additional `providers` by using the `extension` section. I suppose that this is strongly related to dns providers use cases:
+ managing the shoot domain
+ used to create custom dns records via `dnsEntry` within the shoot cluster

Reply:
Good suggestion, +1. The problem, however, is that if we would have the list of additional providers in the provider config of the `dns-service` extension then we cannot reference secrets that contain the credentials. Today, we have no mechanism to allow such references. Gardener needs to copy the provider secrets into the seed and he is the only one that can do it (because the extension only run in the seed).
Still, let's keep it in mind for the future, maybe we can release this one day and can then move it down the extension's provider config.

Reply:
Sure, `secretRef` does not work for `extensions` today. Thanks for pointing that out.